cmd/fetch_repo/module.go

```diff
@@ -45,6 +46,16 @@ func fetchModule(dest, importpath, version, sum string) error {
 		goPath = filepath.Join(goroot, "bin", goPath)
 	}
 
+	// Check whether -modcacherw is supported.
+	// Assume that fetch_repo was built with the same version of Go we're running.
+	modcacherw := false
+	for _, tag := range build.Default.ReleaseTags {
+		if tag == "go1.14" {
+			modcacherw = true
+			break
+		}
+	}
+
```

Comment:
@jayconrod this code seems problematic -- we recently had to change our Go version from the default (a 1.14 build) to 1.13, and all our engineers had to `clean --expunge` because of this code.  We'd get errors like:

```
ERROR: no such package '@com_github_bazelbuild_buildtools//buildifier': failed to fetch com_github_bazelbuild_buildtools: fetch_repo: /private/var/tmp/_bazel_scottb/befbf4792f1e89f7a96261537dac474e-services-go/external/go_sdk/bin/go /private/var/tmp/_bazel_scottb/befbf4792f1e89f7a96261537dac474e-services-go/external/go_sdk/bin/go mod download -json -modcacherw github.com/bazelbuild/buildtools@v0.0.0-20200718160251-b1667ff58f71: flag provided but not defined: -modcacherw
```


Could we just invoke Go here and check the Go binary reported version?

Reply:
I'm not sure that would help. It sounds like the problem was that changing the Go version didn't invalidate the `@bazel_gazelle_go_repository_tools` repo. It probably should.

That said, Go 1.13 is no longer supported by the Go team, so `-modcacherw` should just be on unconditionally. 1.13 no longer receives security updates, so I'd strongly encourage you to upgrade.

Reply:
I think it would help in our case, because the invoked Go binary is actually the right version.  I understanding not wanting to support older versions; but in our case we don't want to conflate go version changes with migration to bazel. :/

A better fix would definitely be to fix any dependency invalidation-- gazelle should get rebuild on go version change.  Is that a gazelle fix, or rules go?

Reply:
It's a Gazelle issue. I've opened #940 to track it separately.